analyzers/src/SonarAnalyzer.CSharp/Metrics/CSharpExecutableLinesMetric.cs

```diff
@@ -112,65 +108,40 @@ private bool FindExecutableLines(SyntaxNode node)
-                        return !HasExcludedCodeAttribute((BaseMethodDeclarationSyntax)node, bmds => bmds.AttributeLists, canBePartial: true);
+                        return !HasExcludedCodeAttribute((BaseMethodDeclarationSyntax)node, x => x.AttributeLists, true);
 
                     case SyntaxKind.PropertyDeclaration:
                     case SyntaxKind.EventDeclaration:
-                        return !HasExcludedCodeAttribute((BasePropertyDeclarationSyntax)node, bpds => bpds.AttributeLists);
+                        return !HasExcludedCodeAttribute((BasePropertyDeclarationSyntax)node, x => x.AttributeLists, false);
 
                     case SyntaxKind.AddAccessorDeclaration:
                     case SyntaxKind.RemoveAccessorDeclaration:
                     case SyntaxKind.SetAccessorDeclaration:
                     case SyntaxKind.GetAccessorDeclaration:
                     case SyntaxKindEx.InitAccessorDeclaration:
-                        return !HasExcludedCodeAttribute((AccessorDeclarationSyntax)node, ads => ads.AttributeLists);
+                        return !HasExcludedCodeAttribute((AccessorDeclarationSyntax)node, ads => ads.AttributeLists, false);
 
                     default:
                         return true;
                 }
             }
 
-            private bool HasExcludedCodeAttribute<T>(T node,
-                                                     Func<T, SyntaxList<AttributeListSyntax>> getAttributeLists,
-                                                     bool canBePartial = false)
-                where T : SyntaxNode
+            private bool HasExcludedCodeAttribute<T>(T node, Func<T, SyntaxList<AttributeListSyntax>> getAttributeLists, bool canBePartial) where T : SyntaxNode
             {
-                var hasExcludeFromCodeCoverageAttribute = getAttributeLists(node)
-                    .SelectMany(attributeList => attributeList.Attributes)
-                    .Any(HasExcludedAttribute);
-
-                if (!canBePartial)
-                {
-                    return hasExcludeFromCodeCoverageAttribute;
-                }
-
-                var nodeSymbol = this.semanticModel.GetDeclaredSymbol(node);
-                switch (nodeSymbol?.Kind)
-                {
-                    case SymbolKind.Method:
-                    case SymbolKind.NamedType:
-                        return hasExcludeFromCodeCoverageAttribute
-                            || nodeSymbol.HasAttribute(KnownType.System_Diagnostics_CodeAnalysis_ExcludeFromCodeCoverageAttribute);
-
-                    default:
-                        return hasExcludeFromCodeCoverageAttribute;
-                }
+                var hasExcludeFromCodeCoverageAttribute = getAttributeLists(node).SelectMany(x => x.Attributes).Any(IsExcludedAttribute);
+                return hasExcludeFromCodeCoverageAttribute || !canBePartial
+                    ? hasExcludeFromCodeCoverageAttribute
```

Comment:
It can be `true` from the initialization. And if it's `true`, then this branch was reached and we need to return `true`. 

If it's `false` and `canBePartial` is `false` (reaching this branch) then we need to return `false`.

Reply:
That's right and therefore I suggest writing `false` instead of `hasExcludeFromCodeCoverageAttribute` because they mean the same thing here and `false` is just easier to comprehend.

Reply:
No worries. We've clarified the situation. 

I will not add the parentheses, the EOLs and formatting take care of that.